src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -322,9 +324,27 @@ public String getRepoOwner() {
      */
     @SuppressWarnings("unchecked")
     @DataBoundSetter
-    public void setTraits(@CheckForNull List<SCMTrait> traits) {
+    public void setTraits(@CheckForNull SCMTrait[] traits) {
```

Comment:
@daspilker Java 8 wasn't happy with the previous signature change you did, so I had to reintroduce the original signature below and added this databound setter in it's place. The UI works with this change, but I don't know if I've broken JobDSL or not :/